src/handlers/push_message.rs

```diff
@@ -168,6 +168,18 @@ pub async fn handler_internal(
         )
     })?;
 
+    // Check request body size shouldn't exceed 4Kb (FCM limitations)
+    if serde_json::to_string(&body)
+        .map_err(|e| (Error::InternalSerializationError(e), None))?
+        .len()
+        > 4096
+    {
+        return Err((
+            Error::PayloadTooLarge("Request body size should not exceed 4Kb".to_string()),
```

Comment:
Why not make that inner string part of the display implementation provided by `thiserror` and you just embed the size of the payload in kb?